src/comm/LinkManager.cc

```diff
@@ -1011,6 +1016,21 @@ void LinkManager::_freeMavlinkChannel(int channel)
 
 void LinkManager::_mavlinkMessageReceived(LinkInterface* link, mavlink_message_t message) {
     link->startMavlinkMessagesTimer(message.sysid);
+
+    // Walk the list of Links. If mavlink forwarding is enabled for a given link then send the data out.
+    for (int i = 0; i < _sharedLinks.count(); i++) {
+        LinkConfiguration* linkConfig = _sharedLinks[i]->getLinkConfiguration();
+
+        bool isUniqueLink = linkConfig->link() != link; // We do not want to send messages back on the link from which they originated
+        bool forwardMavlink = isUniqueLink && linkConfig->isForwardMavlink();
+
+        if (forwardMavlink) {
+            qCDebug(LinkManagerLog) << "Forwarding mavlink packet";
+            uint8_t buffer[MAVLINK_MAX_PACKET_LEN];
+            int len = mavlink_msg_to_send_buffer(buffer, &message);
```

Comment:
This will mess up the message counter. Messages may come from different component IDs and each uses their own counter. Message forwarding must always forward the raw, original message. Not a re-encoded one.

Reply:
Okay I moved the logic, can you let me know if this is correct? I am also wondering if I need to do anything about the "listening port". I'm not entirely sure how sending on UDP works... I thought there was a separate port for sending and receiving.